tasmota/tasmota_xdrv_driver/xdrv_100_modbus_bridge.ino

```diff
@@ -0,0 +1,455 @@
+            ((uint8_t *)&value)[1] = buffer[5 + (count * 4)];
+            ((uint8_t *)&value)[0] = buffer[6 + (count * 4)];
+            ext_snprintf_P(svalue, sizeof(svalue), "%*_f", 10, &value);
+          }
+          else
+          {
+            if ((modbusBridge.type == ModbusBridgeType::mb_int32) ||
+                (modbusBridge.type == ModbusBridgeType::mb_uint32))
+            {
+              uint32_t value = 0;
+              ((uint8_t *)&value)[3] = buffer[3 + (count * 4)]; // Get int values
+              ((uint8_t *)&value)[2] = buffer[4 + (count * 4)];
+              ((uint8_t *)&value)[1] = buffer[5 + (count * 4)];
+              ((uint8_t *)&value)[0] = buffer[6 + (count * 4)];
+              if (modbusBridge.type == ModbusBridgeType::mb_int32)
+                snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%d", value);
+              else
+                snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%u", value);
+            }
+            else if ((modbusBridge.type == ModbusBridgeType::mb_int16) ||
+                     (modbusBridge.type == ModbusBridgeType::mb_uint16))
+            {
+              uint16_t value = 0;
+              ((uint8_t *)&value)[1] = buffer[3 + (count * 2)];
+              ((uint8_t *)&value)[0] = buffer[4 + (count * 2)];
+              if (modbusBridge.type == ModbusBridgeType::mb_int16)
+                snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%d", value);
+              else
+                snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%u", value);
+            }
+            else if ((modbusBridge.type == ModbusBridgeType::mb_int8) ||
+                     (modbusBridge.type == ModbusBridgeType::mb_uint8))
+            {
+              if (modbusBridge.type == ModbusBridgeType::mb_int8)
+                snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%d", (int8_t)(buffer[3 + count]));
+              else
+                snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%u", (uint8_t)(buffer[3 + count]));
+            }
+            else if (modbusBridge.type == ModbusBridgeType::mb_bit8)
+            {
+              uint8_t value = (uint8_t)(buffer[3 + count]);
+              snprintf(svalue, MBR_MAX_VALUE_LENGTH, "%d%d%d%d%d%d%d%d", ((value >> 7) & 1), ((value >> 6) & 1), ((value >> 5) & 1), ((value >> 4) & 1), ((value >> 3) & 1), ((value >> 2) & 1), ((value >> 1) & 1), (value & 1));
+            }
+          }
+          ResponseAppend_P(PSTR("%s"), svalue);
+          if (count < modbusBridge.count - 1)
+            ResponseAppend_P(PSTR(","));
+        }
+
+        ResponseAppend_P(PSTR("]}"));
+        ResponseJsonEnd();
+
+        MqttPublishPrefixTopicRulesProcess_P(RESULT_OR_TELE, PSTR(D_JSON_MODBUS_RECEIVED));
+      }
+    }
+    modbusBridge.deviceAddress = 0;
+    free(buffer);
+  }
+}
+
+/********************************************************************************************/
+
+void ModbusBridgeInit(void)
+{
+  tasmotaModbus = new TasmotaModbus(Pin(GPIO_MBR_RX), Pin(GPIO_MBR_TX));
```

Comment:
Do not ussume users have configured these GPIOs. First check if they did and if not break out. See other drivers how to do this.